homeassistant/components/zha/select.py

```diff
@@ -590,3 +591,60 @@ class AqaraThermostatPreset(ZCLEnumSelectEntity):
     _attribute_name = "preset"
     _enum = AqaraThermostatPresetMode
     _attr_translation_key: str = "preset"
+
+
+class MultitermElectricValveStates(types.enum8):
+    """Electric Valve switch states."""
+
+    Off = 0x00
+    On = 0x01
+
+
+class MultitermHeatingCoolingStates(types.enum8):
+    """Heating / Cooling switch states."""
+
+    Heating = 0x00
+    Cooling = 0x01
+
+
+class MultitermSleepModeStates(types.enum8):
+    """Sleep Mode switch states."""
+
+    Inactive = 0x00
+    Active = 0x01
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names=CLUSTER_HANDLER_BINARY_OUTPUT,
+    models={"ZC0101"},
+)
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="manufacturer",
```

Comment:
what is this for?

Reply:
The idea was to add general handler for binary outputs. Binary Output, as specified in ZCL, has ActiveText, InactiveText, Description and other Attributes. I've tried different approaches to use these Attributes to populate Select Input in the following way: Cluster description to be used as attribute name (which is achieved), ActiveText and InactiveText should've been added as an Enum properties. I couldn't find a proper way to dynamically populate Enum object with properties.
I didn't want to change the "types.enum8" definition to add a method which will create new property during execution of the code as I was unable to analyse further the HA Core code and foresee any arising problems due to the change, due to my lack of knowledge.
Finally I've just decided to define different clusters states in Python as shown. As a result, I needed to add some device filters, so these properties apply only to this particular device and not to every device that has Binary Output Cluster defined.

I am willing to try my best to write a general Binary Output Cluster integration in ZHA, but I will most definitelly need some help from HA Community or Developers. It should render in user interface dynamic Select Input options read from the cluster attributes.

Reply:
Sorry I meant why are there 2 marchers on this? And why are you trying to match just “manufacturer”